pkg/resources/services/service.go

```diff
@@ -46,7 +60,20 @@ func NewClusterIPForMinIO(t *miniov1.Tenant) *corev1.Service {
 
 // ServiceForBucket will return a external name based service
 func ServiceForBucket(t *miniov1.Tenant, bucket string) *corev1.Service {
-	minioPort := corev1.ServicePort{Port: miniov1.MinIOPort, Name: miniov1.MinIOServicePortName}
+	var port int32
+	var name string
+	if t.TLS() {
+		port = 443
+		name = miniov1.MinIOServiceHTTPSPortName
+	} else {
+		port = 80
+		name = miniov1.MinIOServiceHTTPPortName
+	}
+	minioPort := corev1.ServicePort{
+		Port:       port,
+		Name:       name,
+		TargetPort: intstr.FromInt(miniov1.MinIOPort),
```

Comment:
why not use strconv.Itoa()?

Reply:
Using the API framework to keep the types the same: https://godoc.org/k8s.io/api/core/v1#ServicePort